### ISSUE TITLE
Associated Type of Associated Type

### ISSUE THREAD

**Issue by DasLixou** [rust-lang/rust-analyzer#15496]
**rust-analyzer version**: 0.3.1631-standalone (83b3ba1b8 2023-08-19)

**rustc version**: rustc 1.71.1 (eb26296b5 2023-08-03)

**relevant settings**: normal

Minimal reproducable code (different files/mod is important!):
```rust
// runner_front.rs
pub trait RunnerFront {
    type Data;
    type Info: RunnerInfo<Front = Self>;
}
// runner_info.rs
pub trait RunnerInfo {
    type Front: RunnerFront<Info = Self>;
}
// main.rs
struct MyStruct<R>(PhantomData<R>);
impl<R: RunnerInfo> MyTrait for MyStruct<R> {
    fn launch(&self) {
        let preparation = |data: R::Front::Data| {};
    }
}
```
even if I have both traits imported, i still get this error:
```rust
error[E0223]: ambiguous associated type
  --> crates\lagoone_event\src\callback.rs:26:34
   |
26 |         let preparation = |data: R::Front::Data| {};
   |                                  ^^^^^^^^^^^^^^
   |
help: if there were a trait named `Example` with associated type `Data` implemented for `<R as RunnerInfo>::Front`, you could use the fully-qualified path
   |
26 |         let preparation = |data: <<R as RunnerInfo>::Front as Example>::Data| {};
   |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Now this kinda helps because I got it working with
```rust
let preparation = |data: <R::Front as RunnerFront>::Data| {};
```
but this is still confusing and it doesn't give autocorrection.
So is this a bug because I think it could know which trait I need and do auto correction. Is it possible to maybe just completly drop the trait cast stuff and make it work directly?

**Comment by lnicola** [comment]
Your code [is rejected](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=38608e0c799eedabe6d3e4b54793aef9) by the compiler. I don't know why that is the case, but I don't think we should support it when the compiler doesn't.

**Comment by flodiebold** [comment]
This isn't a rust-analyzer bug, if anything it's a language limitation. rust-analyzer could maybe be more helpful in this situation though, like automatically expanding to the full associated type notation.

**Comment by DasLixou** [comment]
> This isn't a rust-analyzer bug, if anything it's a language limitation. rust-analyzer could maybe be more helpful in this situation though, like automatically expanding to the full associated type notation.

Yeah thats what i thought. I was unsure if this behaves to rust-analyzer or rustc

**Comment by ChayimFriedman2** [comment]
As said above this isn't a r-a bug.
